src/main/java/org/launchcode/water_garden_tour/controllers/RegisterController.java

```diff
@@ -44,45 +50,43 @@ private static void setUserInSession(HttpSession session, User user) {
 
     @GetMapping("/register")
     public String displayRegistrationForm(Model model) {
-        model.addAttribute(new User());
+        model.addAttribute(new RegisterDTO());
         model.addAttribute("title", "Registration");
         return "register";
 
     }
 
 
     @PostMapping("/register")
-
-    public String processRegistrationForm(@ModelAttribute @Valid User newUser, Errors errors, Model model, HttpServletRequest request) {
-        if (errors.hasErrors()) {
+    public String processRegistrationForm(@ModelAttribute @Valid RegisterDTO registerDTO, BindingResult result, Model model, Errors errors, HttpServletRequest request) {
+        if (result.hasErrors()) {
             model.addAttribute("title", "Registration");
-            model.addAttribute("user", newUser);
+            return "register";
         }
 
+        Optional<User> existingUser = userRepository.findByUsername(registerDTO.getUsername());
 
-
-        Optional<User> existingUser = userRepository.findByUsername(newUser.getUsername());
-
-        if (existingUser != null) {
+        if (existingUser.isPresent()) {
```

Comment:
Did you mean for this to be not present? Just wondering before it was existingUser != null.

Reply:
I changed existingUser !=null to existingUser.isPresent.  The second one works.

Reply:
just wondering, I wanted to make sure you wouldn't get a NullPointerException from this call. It seems that you will not [per the documentation](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#isPresent--) so resolving this conversation.